File: ACE Hardware/ACE Hardware/InventoryEditCategoryForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using MySql.Data.MySqlClient;

namespace ACE_Hardware
{
    public partial class InventoryEditCategoryForm : Form
    {

        private MySqlConnection conn = new MySqlConnection();
        private string connstring = Program.connstring;

        public InventoryEditCategoryForm()
        {
            InitializeComponent();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            Program.inventoryEditCategoryAddFormInstance = new InventoryEditCategoryAddForm();
            Program.inventoryEditCategoryAddFormInstance.TextBox1.Text = "";
            Program.inventoryEditCategoryAddFormInstance.ShowDialog();
        }

        private void InventoryEditCategoryForm_Load(object sender, EventArgs e)
        {
            refreshcategory();
        }

        private void Button3_Click(object sender, EventArgs e)
        {
            string category;
            try
            {
                category = ListBox1.SelectedItem.ToString();

                switch (MessageBox.Show("Are you sure you want to delete this category(" + category + ")?\nDeleting categories will delete products included in the category","Confirm",MessageBoxButtons.YesNo))
                {
                    case DialogResult.None:
                        break;
                    case DialogResult.OK:
                        break;
                    case DialogResult.Cancel:
                        break;
                    case DialogResult.Abort:
                        break;
                    case DialogResult.Retry:
                        break;
                    case DialogResult.Ignore:
                        break;
                    case DialogResult.Yes:
                        deletecategory(category);
                        break;
                    case DialogResult.No:
                        break;
                    default:
                        break;
                }

            }
            catch (Exception ex)
            {
                MessageBox.Show("Please select category to delete");
            }
        }



        public void refreshcategory()
        {
            ListBox1.Items.Clear();
            Button2.Enabled = false;
            Button3.Enabled = false;

            try
            {
                conn.ConnectionString = connstring;
                conn.Open();

                MySqlCommand comm = new MySqlCommand();
                string query = "SELECT * FROM productcategorytbl ORDER BY productcategoryname";
                MySqlDataReader reader;

                comm.CommandText = query;
                comm.Connection = conn;
                reader = comm.ExecuteReader();


                while (reader.Read())
                    ListBox1.Items.Add(reader["productcategoryname"].ToString());
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
            finally
            {
                conn.Close();
            }
        }

        private void InventoryEditCategoryForm_FormClosed(object sender, FormClosedEventArgs e)
        {
            Program.inventoryFormInstance.refreshcategoryanddatagrid();
            Program.inventoryFormInstance.loadcategory();
        }

        private void ListBox1_SelectedIndexChanged(object sender, EventArgs e)
        {
            Button2.Enabled = true;
            Button3.Enabled = true;
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            Program.inventoryEditCategoryEditFormInstance = new InventoryEditCategoryEditForm();
            Program.inventoryEditCategoryEditFormInstance.TextBox1.Text = "";
            Program.inventoryEditCategoryEditFormInstance.ShowDialog();
        }


        private void deletecategory(string category)
        {
            int categoryindex=0;
            try
            {
                conn.ConnectionString = connstring;
                conn.Open();

                MySqlCommand comm = new MySqlCommand();
                string query = "SELECT * FROM productcategorytbl WHERE productcategoryname = '" + category + "'";
                MySqlDataReader reader;
                comm.Connection = conn;
                comm.CommandText = query;
                reader = comm.ExecuteReader();

                while (reader.Read())
                    categoryindex = System.Convert.ToInt32(reader["productcategoryid"].ToString());
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
            finally
            {
                conn.Close();
            }

            try
            {
                conn.ConnectionString = connstring;
                conn.Open();

                MySqlCommand comm = new MySqlCommand();
                string query = "DELETE FROM productcategorytbl WHERE productcategoryid = " + categoryindex;

                comm.CommandText = query;
                comm.Connection = conn;
                comm.ExecuteNonQuery();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
            finally
            {
                conn.Close();
            }

            try
            {
                conn.ConnectionString = connstring;
                conn.Open();

                MySqlCommand comm = new MySqlCommand();
                string query = "DELETE FROM producttbl WHERE productcategoryid = " + categoryindex;

                comm.CommandText = query;
                comm.Connection = conn;
                comm.ExecuteNonQuery();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
            finally
            {
                conn.Close();
            }

            try
            {
                conn.ConnectionString = connstring;
                conn.Open();

                MySqlCommand comm = new MySqlCommand();
                string query = "INSERT INTO logstbl(username,process,description,dateandtime) VALUES('" + Program.login.username + "','Inventory - Delete Category','Deleted category (" + category + ")',NOW())";

                comm.Connection = conn;
                comm.CommandText = query;
                comm.ExecuteNonQuery();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
            finally
            {
                conn.Close();
            }

            MessageBox.Show("Deleted Successfully");
            refreshcategory();
        }

        private void Button4_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}

File: POS/POS/InventoryEditCategoryForm.vb
Imports MySql.Data.MySqlClient

Public Class InventoryEditCategoryForm

    Dim conn As New MySqlConnection
    Dim connstring As String = "Server=localhost;User Id=root;Password=;Database=POS"

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        InventoryEditCategoryAddForm.TextBox1.Text = ""
        InventoryEditCategoryAddForm.ShowDialog()
    End Sub

    Private Sub InventoryEditCategoryForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        refreshcategory()
    End Sub

    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        Dim category As String
        Try
            category = ListBox1.SelectedItem.ToString()
            Select Case (MsgBox("Are you sure you want to delete this category(" & category & ")?" & vbNewLine & "Deleting categories will delete products included in the category", vbYesNo))
                Case vbYes : deletecategory(category)
            End Select
        Catch ex As Exception
            MsgBox("Please select category to delete")
        End Try
    End Sub

    Public Sub refreshcategory()
        ListBox1.Items.Clear()
        Button2.Enabled = False
        Button3.Enabled = False

        Try
            conn.ConnectionString = connstring
            conn.Open()

            Dim comm As New MySqlCommand
            Dim query As String = "SELECT * FROM productcategorytbl ORDER BY productcategoryname"
            Dim reader As MySqlDataReader

            comm.CommandText = query
            comm.Connection = conn
            reader = comm.ExecuteReader


            While reader.Read
                ListBox1.Items.Add(reader("productcategoryname").ToString)
            End While
        Catch ex As Exception
            MsgBox(ex.ToString())
        Finally
            conn.Close()
        End Try
    End Sub

    Private Sub InventoryEditCategoryForm_Closed(sender As Object, e As EventArgs) Handles Me.Closed
        InventoryForm.refreshcategoryanddatagrid()
        InventoryForm.loadcategory()
    End Sub

    Private Sub ListBox1_SelectedIndexChanged(sender As Object, e As EventArgs) Handles ListBox1.SelectedIndexChanged
        Button2.Enabled = True
        Button3.Enabled = True
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        InventoryEditCategoryEditForm.TextBox1.Text = ""
        InventoryEditCategoryEditForm.ShowDialog()
    End Sub

    Private Sub deletecategory(category As String)
        Dim categoryindex As Integer
        Try
            conn.ConnectionString = connstring
            conn.Open()

            Dim comm As New MySqlCommand
            Dim query As String = "SELECT * FROM productcategorytbl WHERE productcategoryname = '" & category & "'"
            Dim reader As MySqlDataReader
            comm.Connection = conn
            comm.CommandText = query
            reader = comm.ExecuteReader

            While reader.Read
                categoryindex = CInt(reader("productcategoryid").ToString)
            End While
        Catch ex As Exception
            MsgBox(ex.ToString)
        Finally
            conn.Close()
        End Try

        Try
            conn.ConnectionString = connstring
            conn.Open()

            Dim comm As New MySqlCommand
            Dim query As String = "DELETE FROM productcategorytbl WHERE productcategoryid = " & categoryindex

            comm.CommandText = query
            comm.Connection = conn
            comm.ExecuteNonQuery()

        Catch ex As Exception
            MsgBox(ex.ToString)
        Finally
            conn.Close()
        End Try

        Try
            conn.ConnectionString = connstring
            conn.Open()

            Dim comm As New MySqlCommand
            Dim query As String = "DELETE FROM producttbl WHERE productcategoryid = " & categoryindex

            comm.CommandText = query
            comm.Connection = conn
            comm.ExecuteNonQuery()

        Catch ex As Exception
            MsgBox(ex.ToString)
        Finally
            conn.Close()
        End Try

        Try
            conn.ConnectionString = connstring
            conn.Open()

            Dim comm As New MySqlCommand
            Dim query As String = "INSERT INTO logstbl(username,process,description,dateandtime) VALUES('" & LoginForm.username & "','Inventory - Delete Category','Deleted category (" & category & ")',NOW())"

            comm.Connection = conn
            comm.CommandText = query
            comm.ExecuteNonQuery()
        Catch ex As Exception
            MsgBox(ex.ToString)
        Finally
            conn.Close()
        End Try

        MsgBox("Deleted Successfully")
        refreshcategory()
    End Sub

    Private Sub Button4_Click(sender As Object, e As EventArgs) Handles Button4.Click
        Me.Close()
    End Sub
End Class